packages/graphql-codegen-cli/src/utils/watcher.ts

```diff
@@ -23,47 +22,24 @@ function emitWatching(watchDir: string) {
 }
 
 export const createWatcher = (
-  initalContext: CodegenContext,
+  initialContext: CodegenContext,
   onNext: (result: Types.FileOutput[]) => Promise<Types.FileOutput[]>
 ): Promise<void> => {
   debugLog(`[Watcher] Starting watcher...`);
-  let config: Types.Config & { configFilePath?: string } = initalContext.getConfig();
-  const files: string[] = [initalContext.filepath].filter(a => a);
-  const documents = normalizeInstanceOrArray<Types.OperationDocument>(config.documents);
-  const schemas = normalizeInstanceOrArray<Types.Schema>(config.schema);
-
-  // Add schemas and documents from "generates"
-  for (const conf of Object.keys(config.generates).map(filename => normalizeOutputParam(config.generates[filename]))) {
-    schemas.push(...normalizeInstanceOrArray<Types.Schema>(conf.schema));
-    documents.push(...normalizeInstanceOrArray<Types.OperationDocument>(conf.documents));
-    files.push(...normalizeInstanceOrArray(conf.watchPattern));
-  }
-
-  if (documents) {
-    for (const doc of documents) {
-      if (typeof doc === 'string') {
-        files.push(doc);
-      } else {
-        files.push(...Object.keys(doc));
-      }
-    }
-  }
+  let config: Types.Config & { configFilePath?: string } = initialContext.getConfig();
 
-  for (const s of schemas) {
-    const schema = s as string;
-    if (isGlob(schema) || isValidPath(schema)) {
-      files.push(schema);
-    }
-  }
```

Comment:
Most of this logic is moved basically unchanged into `patterns.ts` (see earlier comment)